src/v/pandaproxy/schema_registry/error.cc

```diff
@@ -35,6 +35,20 @@ struct error_category final : std::error_category {
         }
         return "(unrecognized error)";
     }
+    // TODO(Ben): Determine how best to use default_error_condition between
+    // pandaproxy/rest and pandaproxy/schema_registry
+    std::error_condition
+    default_error_condition(int ec) const noexcept override {
+        switch (static_cast<error_code>(ec)) {
+        case error_code::schema_id_not_found:
+        case error_code::subject_not_found:
+        case error_code::subject_version_not_found:
+            return reply_error_code::topic_not_found; // 40401
+        case error_code::schema_invalid:
+            return reply_error_code::unprocessable_entity;
+        }
+        return {};
```

Comment:
what's the `default`error condition? did you mean to return here unprocessable_entity as well?

Reply:
I never really know what to do with a switch over an enum class. In this case:
* `std::error_code{}` is success, without an  `error_category`
* `schema_registry::error_code{}` is success, but isn't named
* `pandaproxy::reply_error_code{}` is success but isn't named
* `std::error_condition{}` is being treated as success under `default_error_condition`.

Everything aligns, the semantics of `error_code` and zero initialisation combine to mean success.

Everything I know, I learnt from Andrzej Krzemieński, who did a great blog series: https://akrzemi1.wordpress.com/2017/08/12/your-own-error-condition/

Reply:
I did the same. I think in the RPC we return an unknown error. 

But sounds good. Was a nit